### PR TITLE
Kendra retriever api

### PR DESCRIPTION
**Pull request by 3coins**
## Description
Replaces [Kendra Retriever](https://github.com/hwchase17/langchain/blob/master/langchain/retrievers/aws_kendra_index_retriever.py) with an updated version that uses the new [retriever API](https://docs.aws.amazon.com/kendra/latest/dg/searching-retrieve.html) which is better suited for retrieval augmented generation (RAG) systems.

**Note**: This change requires the latest version (1.26.159) of boto3 to work. `pip install -U boto3` to upgrade the boto3 version.

cc @hupe1980
cc @dev2049

### CODE REVIEW

**Comment by hupe1980** [comment]
@3coins  I think there is a small error in it: getRelevantsDocuments signature is incorrect. You have to set the params top_k and attribute_filter ​​in the the constructor:  

https://github.com/3coins/langchain/blob/ca7cfe0591bef677e10f97b80cde939e4043c115/langchain/schema.py#L310C1-L314C1

**Comment by 3coins** [comment]
@hupe1980 
Thanks for calling out the change. I will update shortly and resubmit.

**Comment by 3coins** [comment]
@hupe1980 @dev2049 
Submitted updates as per your comment. This should be good to merge now.

**Review by leo-gan — APPROVED** [review]
LGTM Yet, integration tests could be helpful.

**Comment by 3coins** [comment]
@leo-gan 
Thanks for approving. I will be adding tests in the next iteration. There is a Kendra [samples repo](https://github.com/aws-samples/amazon-kendra-langchain-extensions) that is dependent on this change, so this should be merged and released.

File: langchain/retrievers/kendra.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    Example:
+        .. code-block:: python
+
+            retriever = AmazonKendraRetriever(
+                index_id="c0806df7-e76b-4bce-9b5c-d5582f6b1a03"
+            )
+
+    """
+
+    index_id: str
```
Comment (dev2049) [inline langchain/retrievers/kendra.py:158]: note BaseRetriever doesn't inherit from pydantic's BaseModel, so these are all just class attributes

File: langchain/retrievers/aws_kendra_index_retriever.py [hunk 2]
```diff
@@ -1,95 +0,0 @@
-"""Retriever wrapper for AWS Kendra."""
-import re
-from typing import Any, Dict, List
-
-from langchain.schema import BaseRetriever, Document
-
-
-class AwsKendraIndexRetriever(BaseRetriever):
```
Comment (dev2049) [inline langchain/retrievers/aws_kendra_index_retriever.py:8]: happy to let you make this call, but is there any argument to be made to keep this around and add a deprecation warning, for backwards compatibility purposes?

Comment (dev2049) [reply langchain/retrievers/aws_kendra_index_retriever.py:8]: could even make this inherit from AmazonKendraRetriever and in init just map init var names to super() as needed (and get rid of all the methods on this class)

File: langchain/retrievers/kendra.py (hunk 1 above)
Comment (3coins) [reply langchain/retrievers/kendra.py:158]: You are right, we can get rid of these. Let me update this and re-submit.

**Comment by 3coins** [comment]
@dev2049 
Thanks for reviewing. Took care of the class attributes. Let me know if there is anything else I should update.

**Comment by dev2049** [comment]
lgtm, thanks @3coins !

File: docs/extras/modules/data_connection/retrievers/integrations/amazon_kendra_retriever.ipynb
Comment (dev2049) [inline docs/extras/modules/data_connection/retrievers/integrations/amazon_kendra_retriever.ipynb:1]: just noticed there's an index_id left in the notebook — is that intentional?

**Comment by ellisonbg** [comment]
Thanks everyone!

**Pull request merged**